fh_fablib/pre-commit-defaults-with-formatting.yaml

```diff
@@ -0,0 +1,51 @@
+# TODO: doesnt work bc of conflicting black and isort configuration
+
+repos:
+    - repo: local
+      hooks:
+          - id: django-check
+            name: django check
+            entry: sh -c 'venv/bin/python manage.py check'
+            pass_filenames: false
+            language: system
+            always_run: true
+    - repo: https://github.com/asottile/pyupgrade
+      rev: v2.29.1
+      hooks:
+          - id: pyupgrade
+            args: [--py36-plus]
+    - repo: https://github.com/psf/black
+      rev: 21.10b0
+      hooks:
+          - id: black
+    - repo: https://github.com/pycqa/isort
+      rev: 5.10.1
+      hooks:
+          - id: isort
+            args: [--profile=black]
+    - repo: https://github.com/pycqa/flake8
+      rev: 4.0.1
+      hooks:
+          - id: flake8
+            excludes: "migrations/"
```

Comment:
Ich bin nicht ganz glücklich damit; Weil teilweise kann es Sinn machen, Migrations auch zu prüfen, insbesondere wenn es `RunPython` Befehle drin hat oder so.

Mit black formatierte Migrations produzieren ganz selten Fehler bei mir; ist das wirklich notwendig?

Reply:
@matthiask
Finde ich auch unglücklich.

In vielen Projekten werden in der setup.cfg die migrations ausgeklammert (für flake8). Wenn man aber ein `pre-commit run -a` macht, dann wird das ignoriert (weiss nicht warum, vielleicht ein Bug?).

Damit verbunden hab ich das Problem, dass `black` die migrations auch ignoriert.

Reply:
Ah lol, stimmt. Du hast recht, das ist ja der Status Quo... 😄

Reply:
... das exclude aus setup.cfg wird ignoriert, weil pre-commit die Files explizit angibt:
https://flake8.pycqa.org/en/latest/user/using-hooks.html